src/pages/cancel/Cancel.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  BankBottomSheet,
+  Button,
+  InputAccountWrapper,
+  InputBank,
+  Spacing,
+  TextField,
+  Toast,
+} from "@components/commons";
+import { NAVIGATION_STATE } from "@constants/navigationState";
+import { useHeader, useModal, useToast } from "@hooks";
+import React, { useEffect, useState } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { convertingNumber } from "@constants/convertingNumber";
+import * as S from "./Cancel.styled";
+import RadioButton from "./../cancel/components/select/RadioButton";
+import { handleChange, handleBankClick, isFormValid } from "./utils";
+import { useCancelBooking } from "../../hooks/useCancelBooking";
+
+const Cancel = () => {
+  const { setHeader } = useHeader();
+  const { openAlert } = useModal();
+  const { state } = useLocation();
+  const { confirmCancelAction } = useCancelBooking(state.bookerName, state.number, state.password);
+  const navigate = useNavigate();
+  const [isDeposit, setIsDeposit] = useState<boolean | null>(null);
+  const [bankOpen, setBankOpen] = useState(false);
+  const [bankName, setbankName] = useState("");
+  const [accountNumber, setAccountNumber] = useState("");
+  const [accountHolder, setAccountHolder] = useState("");
+
+  const performanceDateArray = state.bookingDetails.performanceDate.split("-");
+  const performanceDataDate = performanceDateArray[2].split("T");
+
+  const handleLeftBtn = () => {
+    navigate(-1);
+  };
+
+  const handleBankOpen = (setOpen: React.Dispatch<React.SetStateAction<boolean>>) => {
+    setOpen((prev) => !prev);
+  };
+
+  useEffect(() => {
+    setHeader({
+      headerStyle: NAVIGATION_STATE.ICON_TITLE,
+      title: "예매 취소하기",
+      leftOnClick: handleLeftBtn,
+    });
+  }, [setHeader]);
+
+  useEffect(() => {
+    if (!state) {
+      openAlert({
+        title: "잘못된 접근입니다.",
+        subTitle: "이전 페이지로 이동합니다.",
+        okText: "확인",
+        okCallback: () => navigate(-1),
+      });
+    }
+  }, [state, openAlert, navigate]);
```

Comment:
p1: 
그리고 url을 통해서 바로 들어올 때, 로그를 찍어보시면 지금 무한 렌더링이 되고있어요!
```suggestion
  }, []);
```

요기의 dependency를 없애면 무한 렌더링이 발생하지 않는데! 이 부분도 왜 그런건지 공부하고 저한테 설명해주세요 ~

Reply:
확인해본 결과 `openAlert`가 계속 새롭게 생성되고 있었습니다!
 `openAlert` 코드를 찾아봤는데
```ts
  const openAlert = (alertProps: Omit<AlertProps, "isOpen">) => {
    setAlert({
      isOpen: true,
      ...alertProps,
    });
  };
```
이렇게 `openModal` 훅 안에 정의되어 있어서 
`const { openAlert } = useModal();` 이렇게 사용하고 있었습니다!

따라서 컴포넌트가 렌더링될 때마다 `openAlert`를 새롭게 생성해서 가져오고 있어서 의존성이 계속 바뀌게 되면서 무한 렌더링 감옥에 갇히게 된거더라구요!!

```ts
const openAlert = useCallback(
    (alertProps: Omit<AlertProps, "isOpen">) => {
      setAlert({
        isOpen: true,
        ...alertProps,
      });
    },
    [setAlert]
  );
```
이렇게 `useCallback`을 감싸서 실행해보니까 의존성에 넣어도 새롭게 생성되고 있지 않아서 변경된다고 판단하지 않아서 렌더링이 반복되지 않는 것을 확인했습니다!

그런데 공용으로 잘 쓰고 있는 훅을 `useCallback`으로 감싸는 것도 헤비하다고 생각했고,
새롭게 생성되지 않도록만 코드를 작성한다면 `useCallback`을 사용하지 않아도 문제가 되지 않기 때문에 제 `useEffect`의 의존성을 수정하는 것이 맞다고 생각했습니다!

의존성을 넣을 때 가끔씩 Warning으로 React는 안에서 사용하는 값들을 의존성으로 넣는 것을 권장한다고 뜨길래,
최근에 의존성에 안에 있는 애들을 넣어줘야겠다고 마음먹었는데 좋지 않은 판단이었던 것 같네요!!
컴포넌트 내에서 해당 값들이 변경될 일이 없다면 굳이 넣지 않는 것이 맞는 것 같습니다!
제가 여기에서 `state`, `openAlert`, `navigate` 3개가 사용되고 있기 때문에 의존성에 넣어줬었는데 잘못된 생각이었네요!
`navigate`는 React Router에서 제공하는 함수라서 참조가 변경되지 않기 때문에 의존성으로 굳이 넣어주지 않아도 된다구 하네용 ㅎㅎ
그리고 현재 `state`는 `useLocation`에서 받아오는 `state`이기 때문에 이 값도 새로 페이지를 접속하지 않는한 변경되지 않기 때문에 의존성을 아예 비워두는 것이 올바른 작성법이겠네요!!
반영 완료했습니다!!